immer/detail/rbts/node.hpp

```diff
@@ -310,16 +316,24 @@ struct node
             });
     }
 
-    static node_t* make_leaf_n(count_t n)
+    static node_t* make_leaf_n_into(void* buffer, std::size_t size, count_t n)
     {
         assert(n <= branches<BL>);
-        auto p = new (heap::allocate(sizeof_leaf_n(n))) node_t;
+        assert(size >= sizeof_leaf_n(n));
+        auto p = new (buffer) node_t;
 #if IMMER_TAGGED_NODE
         p->impl.d.kind = node_t::kind_t::leaf;
 #endif
         return p;
     }
 
+    static node_t* make_leaf_n(count_t n)
+    {
+        assert(n <= branches<BL>);
+        auto m  =heap::allocate(sizeof_leaf_n(n));
```

Comment:
I think the code has not be run through `clang-format`, can you do that?

Reply:
Done